server/trillian_map_server.go

```diff
@@ -187,15 +187,26 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	}
 
 	rootHash, err := smtWriter.CalculateRoot()
+	if err != nil {
+		return nil, fmt.Errorf("CalculateRoot(): %v", err)
+	}
 	newRoot := trillian.SignedMapRoot{
 		TimestampNanos: time.Now().UnixNano(),
 		RootHash:       rootHash,
 		MapId:          req.MapId,
 		MapRevision:    tx.WriteRevision(),
 		Metadata:       req.MapperData,
-		// TODO(al): Actually sign stuff, etc!
-		Signature: &spb.DigitallySigned{},
 	}
+	// Sign the root.
```

Comment:
I like the idea, but I'm not sure how this should actually work. 
`SignObject` needs to operate on the root itself, and then we insert the signature back into the root.  Just worried that if I recreate the root object afterwards, it might be a different object with a different signature.